### PR TITLE
Inline Commenting: Update placement of reply input and add author info header

### PR DESCRIPTION
**Pull request by akasunil**
Part of - https://github.com/WordPress/gutenberg/issues/66377

## What? & Why?

- The reply input box appears above the other reply threads when using the inline commenting feature, whereas it should be at the bottom of every prior thread.

## How?

- This PR adds a user information header over the reply input and changes the reply input's location. Added same user information on new comment also.
- Created `CommentForm` component for re-usability and consistency.

## Testing Instructions

- Enable experimental inline commenting option under `Gutenberg` -> `Experiments`
- Edit any post/page
- Add any block, Click on more menu in block toolbar, Look for comment option.
- It would open the thread in sidebar, Where you can add comment.
- Open up dropdown menu in thread, click reply option, will display reply input.

## Screenshots or screencast 

Screenshot at Oct 29 16-52-51

image

### CODE REVIEW

**Comment by ellatrix** [comment]
Thanks! It would be nice if you could simply click on a comment to show the reply field, instead of having to click the three dots. Can be done separately of course.

**Review by ellatrix — APPROVED** [review]

File: packages/editor/src/components/collab-sidebar/comments.js [hunk 1]
```diff
@@ -402,3 +386,58 @@ function CommentHeader( {
...
+ * Render author information for a comment.
+ *
+ * @param {Object} props        - Component properties.
+ * @param {string} props.avatar - URL of the author's avatar.
+ * @param {string} props.name   - Name of the author.
+ * @param {string} props.date   - Date of the comment.
+ *
+ * @return {JSX.Element} The JSX element representing the author's information.
+ */
+function CommentAuthorInfo( { avatar, name, date } ) {
```
Comment (ellatrix) [inline packages/editor/src/components/collab-sidebar/comments.js]: Can't we reuse this in `AddComment`?

Comment (akasunil) [reply packages/editor/src/components/collab-sidebar/comments.js]: yes, definitely. I'll update the PR.

**Comment by akasunil** [comment]
> It would be nice if you could simply click on a comment to show the reply field, instead of having to click the three dots.

yes. I'm working on this in another PR.

**Comment by ellatrix** [comment]
@akasunil Why does the main comment form not have the text area, while the reply form does? This can probably also be reused and should be the same? The only difference between both forms should be that one has a button that says "Comment", and the other a button that says "Reply", otherwise both of these are the same?

**Comment by akasunil** [comment]
> @akasunil Why does the main comment form not have the text area, while the reply form does? This can probably also be reused and should be the same? The only difference between both forms should be that one has a button that says "Comment", and the other a button that says "Reply", otherwise both of these are the same?

I have extracted comment form to separate component for re-usability and consistency. Now, all ( new comment, edit comment and reply ) form will have the text area for comment input.

**Comment by akasunil** [comment]
@ellatrix Could use your review here.

**Comment by ellatrix** [comment]
It was already approved :)

**Pull request merged**